app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -97,12 +95,8 @@ class ItemDetailPresenter(
             .doOnNext { credentials = it }
             .map { it.toDetailViewModel() }
             .subscribe {
-                if (it.username.isNullOrEmpty()) {
-                    view.showUsernamePlaceholder()
-                } else {
-                    view.showUsername(it.username)
-                }
-                view.updateItem(it)
+                val showUsernamePlaceholder = it.username.isNullOrEmpty()
+                view.updateItem(it, showUsernamePlaceholder)
```

Comment:
I wonder whether the `showUsernamePlaceholder` should really be calculated either in the view model or the view itself?

Reply:
Looking at line 57, it seems that this should be in the view model, as a calculated property/function `hasUsername`.

Reply:
I tried putting this in the view model as well, but like the original commit, this broke the copy ability for some reason. That's why I put it back in the fragment/presenter.